pkg/ucp/backend/service.go

```diff
@@ -19,88 +19,82 @@ package backend
 import (
 	"context"
 	"errors"
-	"fmt"
 	"net/http"
 	"net/url"
 
 	v1 "github.com/radius-project/radius/pkg/armrpc/api/v1"
 	ctrl "github.com/radius-project/radius/pkg/armrpc/asyncoperation/controller"
 	"github.com/radius-project/radius/pkg/armrpc/asyncoperation/worker"
-	"github.com/radius-project/radius/pkg/armrpc/hostoptions"
 	"github.com/radius-project/radius/pkg/sdk"
+	"github.com/radius-project/radius/pkg/ucp"
 	"github.com/radius-project/radius/pkg/ucp/api/v20231001preview"
 	"github.com/radius-project/radius/pkg/ucp/backend/controller/resourcegroups"
 	"github.com/radius-project/radius/pkg/ucp/backend/controller/resourceproviders"
 	"github.com/radius-project/radius/pkg/ucp/datamodel"
-	ucpoptions "github.com/radius-project/radius/pkg/ucp/hostoptions"
 	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
 )
 
-const (
-	UCPProviderName = "System.Resources"
-)
-
 // Service is a service to run AsyncReqeustProcessWorker.
 type Service struct {
 	worker.Service
-
-	config ucpoptions.UCPConfig
+	options *ucp.Options
 }
 
-// NewService creates new service instance to run AsyncRequestProcessWorker.
-func NewService(options hostoptions.HostOptions, config ucpoptions.UCPConfig) *Service {
+// NewService creates new backend service instance to run the async worker.
+func NewService(options *ucp.Options) *Service {
 	return &Service{
+		options: options,
 		Service: worker.Service{
-			ProviderName: UCPProviderName,
-			Options:      options,
+			// Will be initialized later.
+
 		},
-		config: config,
 	}
 }
 
-// Name returns a string containing the UCPProviderName and the text "async worker".
+// Name returns the service name.
 func (w *Service) Name() string {
-	return fmt.Sprintf("%s async worker", UCPProviderName)
+	return "ucp async worker"
 }
 
-// Run starts the service and worker. It initializes the service and sets the worker options based on the configuration,
-// then starts the service with the given worker options. It returns an error if the initialization fails.
+// Run starts the background worker.
 func (w *Service) Run(ctx context.Context) error {
-	if err := w.Init(ctx); err != nil {
-		return err
+	if w.options.Config.Worker.MaxOperationConcurrency != nil {
```

Comment:
You'll notice that there's similar boilerplate across our three control-plane services for initializing the background worker. I don't love the current state, but I need to limit the scope of this PR.